analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ReturnEmptyCollectionInsteadOfNull.cs

```diff
@@ -17,7 +17,7 @@ IEnumerable<string> Strings2
             get
             {
                 return null; // Noncompliant
```

Comment:
I am wondering why did you do this change?  For me both locations are fine as they can be interpreted in different ways like:

- Only null: This null is the problem if you use a different value it will be OK

- Return null: It is a problem that you are returning the null, not just the fact that it is null.

Reply:
The reason is two-fold, code readability of the rule implementation and uniformity on the issues it raises.

When I was reworking the rule to also include `default`, `default(T)` and ternaries, the recursive method that "catches" all the cases for a ternary-return expression is locating all the `WhenTrue/WhenFalse` branches and reporting on every single one of them. 

So if I highlighted the same way as before, `return null` would be entirely highlighted, but so would be a multi-line `return condition ? null : otherCondition ? default : new List<T>`, which would not be uniform. It would also be less readable, i think, especially if you compare it with secondary location highlighting.

You could argue that we could not care about being "uniform" and only highlight the entire thing for `return null` and only the "bad apples" for a big, nested `return {ternary inside a ternary...}`, but the rule would be slightly more conditional and less readable.